components/prism-factor.js

```diff
@@ -0,0 +1,370 @@
+(function (Prism) {
+
+	var comment_inside = {
+		function: {
+			pattern: /(\b)(?:TODO|FIX|FIXME|NOTE|BUG|XX+|HACK|\?\?+|!!+)(?=\b|:)/,
+			lookbehind: true
+		}
+	};
+	var string_inside = {
+		number: /(?:\\[^\s']|%\w)/
+	};
+
+	var factor = {
+
+		'comment': [
+			{
+				// ! single-line exclamation point comments with whitespace after/around the !
+				pattern: /(^|\s)(?:! .*|!$)/,
+				lookbehind: true,
+				inside: comment_inside
+			},
+
+			/* from basis/multiline: */
+			{
```

Comment:
Shouldn't all the below comment patterns be greedy?

Reply:
Being at the top of the file, I don't think it matters, and I can't think of an input that would be classified differently. I've made them greedy for correctness though.